hazelcast/src/test/java/com/hazelcast/map/impl/tx/TxnMapNearCacheInvalidationTest.java

```diff
@@ -74,6 +76,67 @@
     }
 
     @Test
+    public void txn_map_contains_newly_put_key_even_it_is_null_cached_after_addition() throws Exception {
+        final String mapName = "test";
+
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(mapName);
+        nearCacheConfig.setInvalidateOnChange(true);
+
+        Config config = new Config();
+        config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig);
+
+        final HazelcastInstance instance = createHazelcastInstance(config);
+
+        final CountDownLatch waitTxnPut = new CountDownLatch(1);
+        final CountDownLatch waitNullCaching = new CountDownLatch(1);
+
+        final AtomicBoolean keyExistInTxnMap = new AtomicBoolean(false);
+
+        Thread txnThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+
+                Object result = instance.executeTransaction(new TransactionalTask<Object>() {
+                    @Override
+                    public Object execute(TransactionalTaskContext context) throws TransactionException {
+                        TransactionalMap map = context.getMap(mapName);
+                        // 1. First put key into txn map inside txn
+                        map.put(1, 1);
+
+                        waitTxnPut.countDown();
+
+                        assertOpenEventually(waitNullCaching);
+
+                        // 3. After caching key as null inside Near Cache, check to see if key exist for txn
+                        return map.containsKey(1);
+                    }
+                });
+
+                keyExistInTxnMap.set((Boolean) result);
+            }
+        });
+        txnThread.start();
+
+
+        Thread nullCachingThread = new Thread(new Runnable() {
```

Comment:
The test looks like we just need one thread for the `TransactionalTask`. Can't we execute this block directly in the main thread?

Reply:
fixed